compiler/ras/Debug.cpp

```diff
@@ -492,6 +492,16 @@ TR_Debug::getDiagnosticFormat(const char *format, char *buffer, int32_t length)
 bool
 TR_Debug::performTransformationImpl(bool canOmitTransformation, const char * format, ...)
    {
+   int32_t optIndex = _comp->getOptIndex();
+   int32_t firstOptIndex = _comp->getOptions()->getFirstOptIndex();
+   int32_t lastOptIndex = _comp->getOptions()->getLastOptIndex();
+   // An opt that is marked as MustBeDone will always run.
+   // However, we need to omit optional transformations when the opt is supposed
+   // to be prohibited by firstOptIndex and lastOptIndex
+   //
+   if (canOmitTransformation && (optIndex < firstOptIndex || optIndex > lastOptIndex))
+      return false;
```

Comment:
Does returning here interfere with the `TR_CountOptTransformations` code right below? It seems we may not execute `_comp->incOptSubIndex()`, etc. if we return early. Just making sure this is what we want.

Reply:
Yes, this is what we want. If all optional transformations are prohibited by firstOptIndex and lastOptIndex, we don't need to increment sub index and transformation index. Sub index and transformation index are other ways to search bad transformations, but they're only used when the opt index has passed the criteria.